[LG-3973] Add Partnerships data models

orenyk opened the pull request:
Resolves LG-3973

Adds migration and models/specs for Partnerships data.

orenyk commented on app/models/partnerships/iaa_gtc.rb:
```diff
@@ -0,0 +1,14 @@
+  validates :mod_number, presence: true,
+                         numericality: { only_integer: true,
+                                         greater_than_or_equal_to: 0 }
+  # validates :partner_account, foreign_key: true
```
The namespacing of the models is breaking my very naive [validator](https://github.com/18F/identity-idp-config/blob/oyk-partnerships-data/lib/foreign_key_validator.rb) - I'm wondering if we even need ActiveModel validations in the IdP since the foreign key constraints in the DB will catch any issues and I only added the validator in the config repo to make the YAML validation easier to deal with. Curious what you think!

achapm replied:
Interesting point. However, I hope these tables only exist in the IDP some day. I think it makes sense to add the ActiveModel validations and tests here as well for uniformity.

achapm reviewed (approved):
LGTM

orenyk commented:
@jmhooper I changed the namespace and switched the IAA estimated amount columns to decimal types instead of integer to allow for cents. Let me know if there's anything else you think needs to happen and if there's any reason not to just merge this in. Thanks!
